[Bug]: Makefile does not run properly when no "python" is installed on machine (but only "python3", for example)

### Module

Other

### Contact Details

_No response_

### Current Behavior

After `git clone https://github.com/deel-ai/puncc.git` and `cd puncc`, the following action has failed:
> `$ make prepare-dev`
> `python -m venv puncc-dev-env`
> `make: python: Command not found`
> `make: *** [Makefile:17: prepare-dev] Error 127`

Cause: on my local machine (ubuntu via WSL on windows), I do not have a `python` installed, but only `python3` and `python3.X`.

Possible solutions: None trivial, for what I know. 
Using `python3` is a bit better because it is explicit in what we want, but we could have an old (python <= 3.7) version pointed by `python3`.

Remark: I do not know if *not having* a `python` but only `python3` is standard, but since I have it, others could (no major tinkering with python on my machine)

### Expected Behavior

Installation of `prepare-dev` virtual environment with `puncc` and other packages.

### Version

v0.9

### Environment

```markdown
- OS: Ubuntu 20.04.5 LTS (Focal Fossa) on Windows Subsystem for Linux (WSL)
- Python version:
- Packages used version:
```


### Relevant log output

_No response_

### To Reproduce

Within a terminal (linux) without `python`, run:
> `make prepare-dev`

--- a/Makefile
+++ b/Makefile
@@ -14,17 +14,17 @@ help:
 	@echo "       updatetools pre-commit tool"
 
 prepare-dev:
-	python -m venv puncc-dev-env
-	. puncc-dev-env/bin/activate && pip install --upgrade pip
-	. puncc-dev-env/bin/activate && pip install -e .[dev]
+	python3 -m venv puncc-dev-env
+	. puncc-dev-env/bin/activate && python -m pip install --upgrade pip
+	. puncc-dev-env/bin/activate && python -m pip install -e .[dev]
 	. puncc-dev-env/bin/activate && pre-commit install
 	. puncc-dev-env/bin/activate && pre-commit install-hooks
 
 install-user:
-	python -m venv puncc-user-env
-	. puncc-user-env/bin/activate && pip install --upgrade pip
-	. puncc-user-env/bin/activate && pip install -e .[interactive]
-	python -m ipykernel install --user --name=puncc-user-env
+	python3 -m venv puncc-user-env
+	. puncc-user-env/bin/activate && python -m pip install --upgrade pip
+	. puncc-user-env/bin/activate && python -m pip install -e .[interactive]
+	. puncc-user-env/bin/activate && python -m ipykernel install --user --name=puncc-user-env
 
 check_all:
 	. puncc-env/bin/activate && pre-commit run --all-files

